### PR TITLE
app-editors/atom: Version bump to 1.12.2

### PR DESCRIPTION
**Pull request by elprans**
Gentoo-Bug: https://bugs.gentoo.org/show_bug.cgi?id=597442

Package-Manager: portage-2.3.0

### CODE REVIEW

**Comment by gktrk** [comment]
@gentoo/proxy-maint

File: app-editors/atom/atom-1.12.2.ebuild [hunk 1]
```diff
@@ -0,0 +1,400 @@
...
+
+easar() {
+	local asar="${WORKDIR}/$(package_dir asar)/node_modules/asar/bin/asar"
+	echo "asar" $@
+	enode_electron "${asar}" $@ || die
+}
+
+package_dir() {
+	local binmod="${1}" binmod_v
+	eval binmod_v=\${$(tr '[:lower:]' '[:upper:]' <<< ${binmod//-/_}_V)}
```
Comment (gktrk) [inline app-editors/atom/atom-1.12.2.ebuild]: Sorry I missed this one. Can we switch to using ${!foo} type indirections instead?

Comment (elprans) [inline app-editors/atom/atom-1.12.2.ebuild]: Fixed

**Comment by gktrk** [comment]
We have a bit of a mess with the patches. We are missing these

```
 app-editors/atom/files/atom-apm-path.patch | 4 ++--
 app-editors/atom/files/atom-python.patch   | 4 ++--
 2 files changed, 4 insertions(+), 4 deletions(-)

diff --git a/app-editors/atom/files/atom-apm-path.patch b/app-editors/atom/files/atom-apm-path.patch
index 957fed4..7ee7cdb 100644
--- a/app-editors/atom/files/atom-apm-path.patch
+++ b/app-editors/atom/files/atom-apm-path.patch
@@ -1,5 +1,5 @@
---- build/app/src/config-schema.js.orig        2016-05-24 16:59:41.284449076 -0400
-+++ build/app/src/config-schema.js     2016-05-24 17:01:05.685447769 -0400
+--- a/build/app/src/config-schema.js
++++ b/build/app/src/config-schema.js
 @@ -9,6 +9,11 @@
      core: {
        type: 'object',
diff --git a/app-editors/atom/files/atom-python.patch b/app-editors/atom/files/atom-python.patch
index 250ca57..57a475e 100644
--- a/app-editors/atom/files/atom-python.patch
+++ b/app-editors/atom/files/atom-python.patch
@@ -1,5 +1,5 @@
---- ~atom.sh   2015-10-30 11:41:20.090417736 +0100
-+++ atom.sh    2015-10-30 11:42:19.707553644 +0100
+--- a/atom.sh
++++ b/atom.sh
 @@ -92,6 +92,7 @@
    fi
  ```

And it still fails with:

```
 * Applying atom-license-path.patch ...
/var/tmp/portage/app-editors/atom-1.12.2/temp/environment: line 765: /home/gokturk/stuff/gentoo/app-editors/atom/files/atom-license-path.patch: No such file or directory      [ !! ]
```

**Comment by gktrk** [comment]
We might also want to put a hard blocker for [`!!sys-apps/apmd`](http://www.portagefilelist.de/site/query/listPackageFiles/?category=sys-apps&package=apmd&version=3.2.2_p14&do#result)

**Comment by elprans** [comment]
Sorry, I somehow forgot to copy the patches from the overlay.  Now fixed.  apmd blocker added.

**Comment by gktrk** [comment]
Merged in https://github.com/gentoo/gentoo/compare/c6d3f91...60deb5c. Thanks!

I've mangled the commit message to change the Gentoo-Bug tag a bit and rebased the commits before pushing.

**Comment by elprans** [comment]
@gktrk Thanks!

**Pull request closed without merging**